### PR TITLE
Add base href tag support

### PR DESCRIPTION
**Pull request by suplo**
### Purpose/goal of Pull Request (Issue #)

Support `base` tag

> The HTML <base> element specifies the base URL to use for all relative URLs contained within a document. There can be only one <base> element in a document.
> -- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

### CODE REVIEW

File: lib/wovnrb/html_replacers/link_replacer.rb [hunk 1]
```diff
@@ -13,26 +13,70 @@ def initialize(store, pattern, headers)
...
+
+      href= base_tag.get_attribute('href')
+      return href if href =~ /^\//            # absolute path
+      return href if href =~ /^http(s?):\/\// # full url
+
+      abs_path = File.join('/', @headers.dirname, href)
+      strip_relative_path_jumps(abs_path)
+    end
+
+    def strip_relative_path_jumps(path)
```
Comment (a-bougouin) [inline lib/wovnrb/html_replacers/link_replacer.rb]: Is it possible to use `URI` or `Addressable::URI`?
```
[5] pry(main)> URI.join('http://example.com/','main/', '../', 'test')
=> #<URI::HTTP http://example.com/test>
[6] pry(main)> Addressable::URI.join('http://example.com/','main/', '../', 'test')
=> #<Addressable::URI:0x3fe8fd812a38 URI:http://example.com/test>
```

Comment (suplo) [reply lib/wovnrb/html_replacers/link_replacer.rb]: I replaced `File.join` by `Addressable::URI.join` at some places https://github.com/WOVNio/wovnrb/pull/124/commits/162822d84dc5b959bd7fce2aa072a55f908a2986

File: lib/wovnrb/html_replacers/link_replacer.rb [hunk 2]
```diff
@@ -13,26 +13,67 @@ def initialize(store, pattern, headers)
...
+      components = path.split('/') - ['.', '']
+      while components.index('..')
+        if components.index('..').zero?
+          components.shift
+        else
+          components.slice!(components.index('..') - 1, 2)
+        end
+      end
+
+      Addressable::URI.join('/', components.join('/')).to_s
```
Comment (a-bougouin) [inline lib/wovnrb/html_replacers/link_replacer.rb]: I meant: can we use `Addressable::URI.join` to replace home made code:
```
      while components.index('..')
        if components.index('..').zero?
          components.shift
        else
          components.slice!(components.index('..') - 1, 2)
        end
      end
```

Maybe something like:
```
components = path.split('/')
Addressable::URI.join(*components).to_s
```

Comment (suplo) [reply lib/wovnrb/html_replacers/link_replacer.rb]: ah, I misunderstood your comment. Will check. Thanks

Comment (suplo) [reply lib/wovnrb/html_replacers/link_replacer.rb]: Refactor to remove home made code https://github.com/WOVNio/wovnrb/pull/124/commits/946d6db1f1f25367a21f27939dcc5bb2cad7625b

File: lib/wovnrb/html_replacers/link_replacer.rb [hunk 3]
```diff
@@ -58,21 +58,7 @@ def base_href(dom)
...
-      while components.index('..')
-        if components.index('..').zero?
-          components.shift
-        else
-          components.slice!(components.index('..') - 1, 2)
-        end
-      end
-
-      Addressable::URI.join('/', components.join('/')).to_s
+      Addressable::URI.join('/', @headers.dirname, href).to_s
```
Comment (a-bougouin) [inline lib/wovnrb/html_replacers/link_replacer.rb]: :) thanks

File: lib/wovnrb/html_replacers/link_replacer.rb [hunk 4]
```diff
@@ -13,26 +13,53 @@ def initialize(store, pattern, headers)
...
 
         href = node.get_attribute('href')
         next if href =~ /^\s*\{\{.+\}\}\s*$/
         next if href =~ /^\s*javascript:/i
         next if is_file?(href)
-        new_href = lang.add_lang_code(href, @pattern, @headers)
+
+        new_href = href
+        new_href = adjust_link_by_base(new_href, base_url) if base_url
+        new_href = lang.add_lang_code(new_href, @pattern, @headers)
```
Comment (tim-wovn) [inline lib/wovnrb/html_replacers/link_replacer.rb]: Is this line related to this PR?

File: lib/wovnrb/html_replacers/link_replacer.rb [hunk 5]
```diff
@@ -13,26 +13,53 @@ def initialize(store, pattern, headers)
...
       end
     end
 
+    private
+
+    def adjust_link_by_base(href, base_url)
+      return href if href =~ /^\//            # absolute path
+      return href if href =~ /^http(s?):\/\// # full url
+
+      File.join(base_url, href)
```
Comment (a-bougouin) [inline lib/wovnrb/html_replacers/link_replacer.rb]: Maybe we should use `Addressable::URI.join` here too?

Comment (suplo) [reply lib/wovnrb/html_replacers/link_replacer.rb]: I tried. It does not work
